assets/src/components/animation-order-picker.js

```diff
@@ -0,0 +1,110 @@
+/**
+ * WordPress dependencies
+ */
+import { __, sprintf } from '@wordpress/i18n';
+import { Dashicon, BaseControl, Button, Dropdown, NavigableMenu } from '@wordpress/components';
+import { BlockIcon } from '@wordpress/editor';
+import { Fragment } from '@wordpress/element';
+
+function ButtonContent( { option, displayIcon = true } ) {
+	const { label: name, block, blockType } = option;
+
+	if ( ! block ) {
+		return name;
+	}
+
+	let label = block.clientId;
+
+	// Todo: Cover more special cases if needed.
+	switch ( block.name ) {
+		case 'core/image':
+			if ( block.attributes.url ) {
+				label = block.attributes.url.slice( block.attributes.url.lastIndexOf( '/' ) );
+			}
+
+			break;
+		case 'amp/amp-story-text':
+			const content = block.originalContent ? block.originalContent.replace( /<[^<>]+>/g, ' ' ) : '';
```

Comment:
Maybe this would need to make use of `block.attributes.content` instead of `block.originalContent`? Perhaps the `originalContent` is not updated and shows the original value?

Reply:
What did I say about the forest and the trees?! That was it 🤦‍♂️ 

Thanks @miina 🎉